### PR TITLE
extract `prettier-config-yoshi` package

### PR DESCRIPTION
**Pull request by bodia-uz**
### 🔦 Summary

To share one prettier config between different projects, we need to extract yoshi prettier config to a separate package (`prettier-config-yoshi`).

It will allow us to use it in existing `eslint-config-yoshi-base`, `tslint-config-yoshi-base` projects, and other projects that want to be aligned with yoshi.

### 🗺️ Test Plan

its just config file. Not sure if we need to test it in some way.

### CODE REVIEW

File: packages/prettier-config-yoshi/package.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "prettier-config-yoshi",
+  "version": "4.4.3",
+  "description": "prettier configuration for projects using yoshi",
+  "main": "index.js",
+  "keywords": [
+    "prettier",
+    "yoshi"
+  ],
+  "author": "Ronen Amiel & Ran Yitzhaki",
```
Comment (zemlanin) [inline packages/prettier-config-yoshi/package.json]: Bohdan, don't be shy 🙃

Comment (bodia-uz) [reply packages/prettier-config-yoshi/package.json]:
```suggestion
  "author": "Ronen Amiel & Ran Yitzhaki & Bohdan Lyzanets",
```
;)

File: packages/prettier-config-yoshi/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# prettier configuration for yoshi based projects
+
+`prettier.config.js` or `.prettierrc.js`:
+
+```
+const prettierConfig = require('prettier-config-yoshi');
+
+module.exports = {
+  ...prettierConfig,
+  semi: false
```
Comment (yavorsky) [inline packages/prettier-config-yoshi/README.md]: I think the idea of having a common place for prettier config is to just reuse it, not extend. I mean, prettier configuration is not to big just less than 10 options. I won't encourage people to extend it in docs.
We are extending eslint-config using *-base* approach just b/c we want to add new rules (specific to project environment), not override existing ones.

File: packages/prettier-config-yoshi/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+eslint config:
+
+```js
+const prettierConfig = require("prettier-config-yoshi");
+
+module.exports = {
+  plugins: ["prettier"],
+  rules: {
+    "prettier/prettier": ["warn", prettierConfig]
```
Comment (yavorsky) [inline packages/prettier-config-yoshi/README.md]: I would also add to docs that if you already have `.prettierrc` or other prettier configuration in the project, you don't need to pass prettierConfig to eslint rule.

File: packages/prettier-config-yoshi/index.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  singleQuote: true,
+  trailingComma: 'all',
```
Comment (yavorsky) [inline packages/prettier-config-yoshi/index.js]: Not sure about this, but maybe adding other options that are set by default (like printWidth: 80, bracketSpacing: true) will help to highlight other fields in editor during import.

File: packages/prettier-config-yoshi/README.md (hunk 2 above)
Comment (bodia-uz) [reply packages/prettier-config-yoshi/README.md]: I copied a few examples from prettier docs and adapt the, but you a right, better to not provoke people to do it=)

File: packages/prettier-config-yoshi/index.js (hunk 4 above)
Comment (bodia-uz) [reply packages/prettier-config-yoshi/index.js]: Maybe. @ranyitz what do you think?

File: packages/prettier-config-yoshi/README.md (hunk 3 above)
Comment (bodia-uz) [reply packages/prettier-config-yoshi/README.md]: Updated this section. Now it's less detailed and recommends the user to use `eslint/tslint-config-yoshi-base`

**Comment by bodia-uz** [comment]
@ranyitz the main purpose of this PR is to allow Wix projects that are not under Yoshi, to be aligned with Yoshi prettier config without thinking what is inside this config. We planned to use it in our project.
For now, we just added the same config in our repo. It's ok for us and we can live with it, or we can import it from `prettier-config-yoshi` if you find it reasonable to add it.

**Comment by ranyitz** [comment]
@bodia-uz I understand. 

> allow Wix projects that are not under Yoshi, to be aligned with Yoshi prettier config without thinking about what is inside this config

If you'll be aligned with `eslint-config-yoshi` you'll be aligned with the prettier config as well.

____________

What's the use-case for a non Yoshi project in Wix?

**Comment by zemlanin** [comment]
> What's the use-case for a non Yoshi project in Wix?

Projects not _(yet?)_ on yoshi, like https://github.com/wix-private/santa-editor

**Comment by bodia-uz** [comment]
@ranyitz as Anton said we want to use in our project `santa-editor`, which is not under yoshi now, and not ready yet, to use `eslint-config-yoshi`.

**Comment by ranyitz** [comment]
@bodia-uz @zemlanin If that's the case, I would go with duplicating
```
singleQuote: true,	
trailingComma: 'all',
```

Paying the price of the duplication in this case is lower than the price of maintaining another package.

**Comment by zemlanin** [comment]
I'm not much in favor of having a package with two-keys-object, but duplicating doesn't seem great to me also…

Would using `require('packages/eslint-config-yoshi-base').rules['prettier/prettier']` in `santa-editor` be _so_ bad?..

**Comment by ranyitz** [comment]
@zemlanin that's totally ok
```
require('eslint-config-yoshi-base').rules['prettier/prettier']
```

In my opinion (which I understand can be controversial) repeating yourself in some cases is totally ok, and just a matter of tradeoff. It depends on how many times the code is repeated, how much code is repeated, how much this code is likely to be changed in the future, how much effort it takes to not repeat it and more.

In this case, both solutions seems fine to me, but managing another package for prettier config is not something I want to maintain.

**Comment by bodia-uz** [comment]
Cool.
We will go with duplication for now. Then (when ready) migrate to `eslint --fix` using `eslint-config-yoshi` config.

Thanks for the discussion.

**Pull request closed without merging**